Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Builder
-FROM node:current as builder
+FROM --platform=$BUILDPLATFORM node:current as builder
```

Comment:
Shouldn't this be set to some default to beable to build the Image manually?

Reply:
That *is* a default to be able to build the image manually. By default it will be built for the platform that the builder is running on. If you're building it on an amd64 machine, it'll build an amd64 image. If you're building on an armv7 machine, it'll build an armv7 image.

For a regular build for a single architecture, this is functionally identical to the way it was before I changed it.

Where this makes a *difference*, is that if this was being built for multiple architectures without `--platform=$BUILDPLATFORM`, the `builder` stage would be run once for *each* architecture being built for. So if you built for amd64, arm64, and armv7, you'd end up running the builder 3 times, and because the application is not platform-specific, you'd build three identical copies of the application unnecessarily.
With this change I'm proposing, the builder will only be run *once*, even though the second stage of the Dockerfile will be run once *per* architecture (and that's the fast stage, saving a bunch of CPU time).

Does this make sense?

Reply:
Didn't know it sets/ignores this by default even when not using buildx.

Reply:
Yup:

> The optional `--platform` flag can be used to specify the platform of the image in case `FROM` references a multi-platform image. For example, `linux/amd64`, `linux/arm64`, or `windows/amd64`. By default, the target platform of the build request is used. Global build arguments can be used in the value of this flag, for example [automatic platform ARGs](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) allow you to force a stage to native build platform (`--platform=$BUILDPLATFORM`), and use it to cross-compile to the target platform inside the stage.

(From https://docs.docker.com/engine/reference/builder/#from)